cmd/cstorvolumeclaim/controller.go

```diff
@@ -0,0 +1,404 @@
+	startTime := time.Now()
+	glog.V(4).Infof("Started syncing cstorvolumeclaim %q (%v)", key, startTime)
+	defer func() {
+		glog.V(4).Infof("Finished syncing cstorvolumeclaim %q (%v)", key, time.Since(startTime))
+	}()
+
+	// Convert the namespace/name string into a distinct namespace and name
+	namespace, name, err := cache.SplitMetaNamespaceKey(key)
+	if err != nil {
+		runtime.HandleError(fmt.Errorf("invalid resource key: %s", key))
+		return nil
+	}
+
+	// Get the cvc resource with this namespace/name
+	cvc, err := c.cvcLister.CStorVolumeClaims(namespace).Get(name)
+	if k8serror.IsNotFound(err) {
+		runtime.HandleError(fmt.Errorf("cstorvolumeclaim '%s' has been deleted", key))
+		return nil
+	}
+	if err != nil {
+		return err
+	}
+	cvcCopy := cvc.DeepCopy()
+	err = c.syncCVC(cvcCopy)
+	return err
+}
+
+// enqueueCVC takes a CVC resource and converts it into a namespace/name
+// string which is then put onto the work queue. This method should *not* be
+// passed resources of any type other than CStorVolumeClaims.
+func (c *CVCController) enqueueCVC(obj interface{}) {
+	var key string
+	var err error
+	if key, err = cache.MetaNamespaceKeyFunc(obj); err != nil {
+		runtime.HandleError(err)
+		return
+	}
+	c.workqueue.Add(key)
+
+	/*	if unknown, ok := obj.(cache.DeletedFinalStateUnknown); ok && unknown.Obj != nil {
+			obj = unknown.Obj
+		}
+		if cvc, ok := obj.(*apis.CStorVolumeClaim); ok {
+			objName, err := cache.DeletionHandlingMetaNamespaceKeyFunc(cvc)
+			if err != nil {
+				glog.Errorf("failed to get key from object: %v, %v", err, cvc)
+				return
+			}
+			glog.V(5).Infof("enqueued %q for sync", objName)
+			c.workqueue.Add(objName)
+		}
+	*/
+}
+
+// synCVC is the function which tries to converge to a desired state for the
+// CStorVolumeClaims
+func (c *CVCController) syncCVC(cvc *apis.CStorVolumeClaim) error {
+	// create is a wrapper function that calls the actual function to create pool as many time
+	// as the number of pools need to be created.
+	//	var newCVCLease Leaser
+	//	newCVCLease = &Lease{cvc, cvcLeaseKey, c.clientset, c.kubeclientset}
+	//	err := newCVCLease.Hold()
+	//	if err != nil {
+	//		return errors.Wrapf(err, "Could not acquire lease on cvc object")
+	//	}
+	//	glog.V(4).Infof("Lease acquired successfully on CStorVolumeClaims %s ", spc.Name)
+	//	defer newCVCLease.Release()
+
+	// CStor Volume Claim should be deleted. Check if deletion timestamp is set
+	// and remove finalizer.
+	if isClaimDeletionCandidate(cvc) {
+		glog.Infof("syncClaim: remove finalizer for CStorVolumeClaimVolume [%s]", cvc.Name)
+		return c.removeClaimFinalizer(cvc)
+	}
+
+	volName := cvc.Name
+	if volName == "" {
+		// We choose to absorb the error here as the worker would requeue the
+		// resource otherwise. Instead, the next time the resource is updated
+		// the resource will be queued again.
+		runtime.HandleError(fmt.Errorf("%+v: cvc name must be specified", cvc))
+		return nil
+	}
+
+	nodeID := cvc.Publish.NodeId
```

Comment:
How is this nodeID being used? Why is there a dependency on this? Can the reason be listed a comment above this line.

Reply:
Added